package dmModelDDF;

import "ddf/ddf_extensions.proto";
import "ddf/ddf_math.proto";

option java_package = "com.dynamo.model.proto";
option java_outer_classname = "Model";

message ModelDesc
{
    required string name        = 1;
    required string mesh        = 2 [(resource)=true];
    required string material    = 3 [(resource)=true];
    repeated string textures    = 4 [(resource)=true];
}

message SetVertexConstant
{
    required uint32         register    = 1;
    required dmMath.Vector4 value       = 2;
}

message ResetVertexConstant
{
    required uint32 register = 1;
}

message SetFragmentConstant
{
    required uint32         register    = 1;
    required dmMath.Vector4 value       = 2;
}

message ResetFragmentConstant
{
    required uint32 register = 1;
}

message SetTexture
{
    required uint64 texture_hash = 1;
    required uint32 texture_unit = 2;
}
